src/gurobi_optimods/min_cost_flow.py

```diff
@@ -171,27 +181,40 @@ def min_cost_flow_networkx(G, *, create_env):
     logger.info(
         f"Solving min-cost flow with {len(G.nodes)} nodes and {len(G.edges)} edges"
     )
-
     with create_env() as env, gp.Model(env=env) as model:
+        use_multigraph = isinstance(G, nx.MultiGraph)
+
+        G = nx.MultiDiGraph(G)
+
         edges, capacities, costs = gp.multidict(
-            {(i, j): [d["capacity"], d["cost"]] for i, j, d in G.edges(data=True)}
+            {
+                (i, j, idx): [d["capacity"], d["cost"]]
+                for i, j, idx, d in G.edges(data=True, keys=True)
+            }
         )
+
         nodes = list(G.nodes(data=True))
         x = {
-            (i, j): model.addVar(
-                name=f"flow[{i},{j}]",
-                ub=capacities[i, j],
-                obj=costs[i, j],
+            (i, j, idx): model.addVar(
+                name=f"flow[{i},{j},{idx}]",
+                ub=capacities[i, j, idx],
+                obj=costs[i, j, idx],
             )
-            for i, j in edges
+            for i, j, idx in edges
         }
 
         flow_constrs = {}
         for n, data in nodes:
+            predecessors = [
+                key for key in x.keys() if key[0] in G.predecessors(n) and key[1] == n
+            ]
+            successors = [
+                key for key in x.keys() if key[0] == n and key[1] in G.successors(n)
+            ]
```

Comment:
Could we do this a bit more efficiently? Maybe collect a mapping first from node -> incoming and outgoing edge vars, then build the constraints via a lookup?

Reply:
Good point [44def1a](https://github.com/Gurobi/gurobi-optimods/pull/157/commits/44def1a6bb15eff09c69d3b6f3fb3559463a5186) uses another graph function which resolves this and is similar to what we were doing previously

Reply:
Please let me know if I need to make any changes to get this comment resolved. From looking at the commit history, it seems this has been resolved.

Reply:
Thanks @torressa, looks good. @jsfreischuetz we're all set, the delay is just due to time zone ping pong since I'm in Australia. Merging now :-)